### ISSUE TITLE
使用 numpy 搭建简单神经网络基础框架

### ISSUE THREAD

**Issue by NayukiChiba** [NayukiChiba/Simple-Neural-Network#1]
## 背景

基于 `numpy` 从零实现一个简单神经网络框架，并通过根目录 `main.py` 统一调度训练流程。

优先实现多层感知机（MLP）的核心链路。

## 目标

- 使用 `numpy` 实现简单神经网络的完整训练流程
- 支持基础分类任务与基础回归任务
- 使用根目录 `datasets/` 存放数据集
- 使用根目录 `main.py` 作为统一入口
- 保持模块职责清晰，便于后续扩展

## 范围

包含以下内容：

- 数据集生成与加载
- 线性层与激活函数
- 前向传播与反向传播
- 分类损失函数与回归损失函数
- SGD 优化器
- 基础训练循环
- 模型参数保存与加载

暂不包含以下内容：

- 自动求导
- CNN / RNN
- GPU 支持
- Adam
- BatchNorm
- Dropout

## 推荐目录结构

```text
Simple-Neural-Network/
├── datasets/
│   ├── xor/
│   ├── spiral/
│   └── sine/
├── main.py
├── src/
│   ├── __init__.py
│   └── nn/
│       ├── data/
│       ├── layers/
│       ├── losses/
│       ├── models/
│       ├── optimizers/
│       ├── persistence/
│       └── training/
└── tests/
```

### 阶段一：数据链路

- [x] 设计统一数据格式，优先使用 `.npz`
- [x] 实现 `src/nn/data/datasetGenerator.py`
- [x] 实现 `src/nn/data/datasetLoader.py`
- [x] 生成 `xor` 数据集
- [x] 生成 `spiral` 数据集
- [x] 生成 `sine` 数据集
- [x] #3
- [x] 在命令行中输出数据形状和任务类型
- [x] #5

### 阶段二：网络层

- [x] #7
- [x] #9
- [x] #10 

### 阶段三：损失函数与模型容器

- [x] #11
- [x] #12
- [x] #13

### 阶段四：优化器与训练流程

- [x] #20
- [x] #22
- [x] #24
- [x] 支持 mini-batch 训练
- [x] 支持分类准确率统计
- [x] 支持回归误差统计
- [x] 确认训练过程中损失能够下降

### 阶段五：模型持久化

- [x] #26
- [x] 支持模型参数保存
- [x] 支持模型参数加载
- [x] 验证加载前后预测结果一致

### 阶段六：测试与收尾

- [x] 建立 `tests/` 目录
- [x] 添加数据加载测试
- [x] 添加前向传播测试
- [x] 添加损失函数测试
- [x] 添加简单训练流程测试
- [x] 补充 README 中的运行说明

## 验收标准

- [x] 可以通过 `python main.py --task xor` 成功加载并训练
- [x] 可以通过 `python main.py --task spiral` 成功加载并训练
- [x] 可以通过 `python main.py --task sine` 成功加载并训练
- [x] 代码结构与模块职责清晰
- [x] 数据、模型、训练流程彼此解耦
- [x] 第一版具备完整的神经网络主流程

## 备注

建议按以下顺序推进：

1. 数据生成与加载
2. `main.py` 数据调度
3. 线性层与激活函数
4. 损失函数
5. 顺序模型
6. 优化器
7. 训练循环
8. 模型保存与加载
9. 测试补全
